import_token.go

```diff
@@ -133,6 +134,52 @@ func encryptNonce(key, value, iv string) (string, string, error) {
 	return base64.StdEncoding.EncodeToString(cipherText), iv, nil
 }
 
+// EncryptNonceWithCBC encrypts the nonce using the user's key-material
+// with CBC encrypter, it requires padding, here pkcs7 padding is used.
+// This is needed for Hyper Protect Crypto Services, since it supports only
+// CBC Encryption.
+func EncryptNonceWithCBC(key, value, iv string) (string, string, error) {
+	keyMat, err := base64.StdEncoding.DecodeString(key)
+	if err != nil {
+		panic(err)
+	}
+
+	nonce, err := base64.StdEncoding.DecodeString(value)
+	if err != nil {
+		panic(err)
+	}
+
+	block, err := aes.NewCipher(keyMat)
+	if err != nil {
+		panic(err)
+	}
+
+	// PKCS7 Padding
+	paddingLength := aes.BlockSize - len(nonce)%aes.BlockSize
+	paddingBytes := []byte{byte(paddingLength)}
+	paddingText := bytes.Repeat(paddingBytes, paddingLength)
+	nonce = append(nonce, paddingText...)
+
+	var newIv []byte
+	if iv != "" {
+		newIv = []byte(iv)
+	} else {
+		newIv = make([]byte, aes.BlockSize)
+	}
+
+	// Generate an IV to achieve semantic security
+	if _, err := io.ReadFull(rand.Reader, newIv); err != nil {
```

Comment:
Uh... I think this needs to be in the else block above. You are always overriding the passed in IV with random bytes here...